pkg-manager/plugin-commands-installation/test/install.ts

```diff
@@ -52,3 +53,124 @@ test('install with no store integrity validation', async () => {
 
   expect(fs.readFileSync('node_modules/is-positive/readme.md', 'utf8')).toBe('modified')
 })
+
+describe('install injected dependencies', () => {
+  for (const {
+    title,
+    withLockfile,
+  } of [
+      {
+        title: 'with shared lockfile',
+        withLockfile: true,
+      }, {
+        title: 'with individual lockfiles',
+        withLockfile: false,
+      },
+    ]) {
+    test(title, async () => {
+      preparePackages([
+        {
+          location: '.',
+          package: {
+            name: 'root',
+            version: '1.0.0',
+          },
+        },
+        {
+          name: 'project-1',
+          version: '1.0.0',
+          dependencies: {
+            'project-2': 'workspace:*',
+          },
+          devDependencies: {
+            'project-3': 'workspace:*',
+          },
+          dependenciesMeta: {
+            'project-2': {
+              injected: true,
+            },
+          },
+        },
+        {
+          name: 'project-2',
+          version: '1.0.0',
+          devDependencies: {
+            'project-3': 'workspace:*',
+          },
+          publishConfig: {
+            exports: {
+              foo: 'bar',
+            },
+          },
+        },
+        {
+          location: 'nested-packages/project-3',
+          package: {
+            name: 'project-3',
+            version: '1.0.0',
+          },
+        },
+      ])
+
+      const { allProjects, allProjectsGraph, selectedProjectsGraph } = await readProjects(process.cwd(), [])
+
+      const lockfileDir = withLockfile ? process.cwd() : undefined
+
+      await install.handler({
+        ...DEFAULT_OPTS,
+        allProjects,
+        allProjectsGraph,
+        dir: process.cwd(),
+        lockfileDir,
+        selectedProjectsGraph,
+        workspaceDir: process.cwd(),
+      })
+
+      const [
+        originalProject2,
+        injectedProject2,
+      ] = await Promise.all([
+        fs.promises.readFile('project-2/package.json', 'utf8'),
+        fs.promises.readFile('project-1/node_modules/project-2/package.json', 'utf8'),
+      ])
+
+      expect(
+        JSON.parse(originalProject2)
+      ).toEqual({
+        name: 'project-2',
+        version: '1.0.0',
+        devDependencies: {
+          'project-3': 'workspace:*',
+        },
+        publishConfig: {
+          exports: {
+            foo: 'bar',
+          },
+        },
+      })
+
+      expect(
+        JSON.parse(injectedProject2)
+      ).toEqual({
+        name: 'project-2',
+        version: '1.0.0',
+        devDependencies: {
+          'project-3': '1.0.0',
+        },
+        exports: {
```

Comment:
An important question is whether this package.json is expected to be "publishable".

I believe existing (pre-regression) this would _not_ be, but post this PR: https://github.com/pnpm/pnpm/pull/6943 we would expect to to be

If we _do not_ want it to be "publishable", then that transform should be moved out of the directory fetcher (and simplifies this regression).